server/api/users.js

```diff
@@ -53,3 +53,68 @@ router.get('/:id/latestguesses', async (req, res, next) => {
 		next(error)
 	}
 })
+
+const getRandomInt = max => {
+	return Math.floor(Math.random() * max)
+}
+
+const completedPrevGameOrStartNewGame = async (req, res, next) => {
+	try {
+		let latestWordle = await WordleGame.findAll({
+			where: { userId: req.params.id },
+			include: AcceptedGuess
+		})
+		latestWordle ? (latestWordle = latestWordle[latestWordle.length - 1]) : null
+		if (!latestWordle || latestWordle.acceptedGuesses.length === 5) {
+			next()
+		} else {
+			console.log('not finished game yet')
+		}
+	} catch (error) {
+		next(error)
+	}
+}
+
+//create a new target word and asign that to a new game - send back the new target word to client
+router.get(
+	'/:id/newTargetWord',
+
+	completedPrevGameOrStartNewGame,
+	async (req, res, next) => {
+		try {
+			const randomWord = await AcceptedWord.findByPk(getRandomInt(10656))
+			const targetWord = await TargetWord.create({ content: randomWord.content })
+			const newWordleGame = await WordleGame.create({
+				targetWordId: targetWord.id,
+				userId: req.params.id
+			})
+			res.json({ targetWord })
+		} catch (error) {
+			next(error)
+		}
+	}
+)
+
+//adding player's guesses to a wordle game in database as long as they have not finish the game
+router.post('/:id/latestguesses', async (req, res, next) => {
```

Comment:
Is there a way we could pass the wordle game id down for this post route? Like if we have already pulled the latest game we should have access to the wordle game id. That would ensure this post route is 100% accurate!